packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -341,4 +342,28 @@ export class VueElement extends BaseClass {
       })
     }
   }
+
+  private _getStylesRecursively(
```

Comment:
This only addresses styles for locally registered non-async components. Async or global components would still have no styles.

Reply:
I consider a web component to be an isolated and self-contained component in the shadow DOM, which is fully loaded and registered as a module.
I would be concerned if code was dynamically reloaded in the shadow DOM, so I don't see the need for async components.